appender.go

```diff
@@ -179,18 +176,29 @@ func (a *Appender) Close(ctx context.Context) error {
 	defer a.mu.Unlock()
 	select {
 	case <-a.closed:
+		return a.errgroup.Wait()
 	default:
-		close(a.closed)
+	}
+	close(a.closed)
 
-		// Cancel ongoing flushes when ctx is cancelled.
-		ctx, cancel := context.WithCancel(ctx)
-		defer cancel()
-		go func() {
-			defer a.cancelErrgroupContext()
-			<-ctx.Done()
-		}()
+	// Cancel ongoing flushes when ctx is cancelled.
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	go func() {
+		defer a.cancelErrgroupContext()
+		<-ctx.Done()
+	}()
+
+	if err := a.errgroup.Wait(); err != nil {
+		return err
 	}
-	return a.errgroup.Wait()
+	close(a.available)
+	for bi := range a.available {
+		if err := a.flush(context.Background(), bi); err != nil {
+			return fmt.Errorf("failed to flush events on close: %w", err)
+		}
+	}
+	return nil
```

Comment:
Returning early will cause the non-empty `bulkIndexers` to not be flushed. What do you think about the snippet below?
```go
	// Wait until all active indexers have been flushed.
	err := a.errgroup.Wait()
	close(a.available)
	for bi := range a.available {
		a.errgroup.Go(func() error {
			if e := a.flush(context.Background(), bi); e != nil {
				return fmt.Errorf("failed to flush events on close: %w", e)
			}
			return nil
		})
	}
	return errors.Join(err, a.errgroup.Wait())
```

Reply:
Could you open a follow up issue then to tackle this? With adding retry behavior we are moving more towards reliability vs. dropping events, so the ask to change the behavior for flushing non empty bulk indexers seems totally fair.